xnmt/encoder.py

```diff
@@ -36,17 +36,27 @@ def transduce(self, embed_sent):
     """
     raise NotImplementedError('Unimplemented transduce for class:', self.__class__.__name__)
 
+  def get_final_state(self):
+    """ Return the state that represents the transduced sequence """
+    return None
```

Comment:
Our current behavior is:
* `Retriever`: never call `get_final_state` so behavior is irrelevant.
* `Translator`: call `get_final_state` and implement special error handling that throws an error when we receive `None`.

What if we implement a similar class `Tagger` that also calls `get_final_state`. Then we'd have to implement the same error handling in `Tagger` as well. Thus, error handling should be done at the source of the error (here) instead.